db/external_sst_file_ingestion_job.cc

```diff
@@ -390,34 +390,57 @@ Status ExternalSstFileIngestionJob::AssignLevelForIngestedFile(
     SuperVersion* sv, IngestedFileInfo* file_to_ingest, bool* overlap_with_db) {
   *overlap_with_db = false;
 
-  Arena arena;
-  ReadOptions ro;
-  ro.total_order_seek = true;
-
   Status status;
   int target_level = 0;
   auto* vstorage = cfd_->current()->storage_info();
+
+  // Always go to last level, fail if the file overlaps with existing files.
+  if (ingestion_options_.ingest_behind) {
```

Comment:
It would be better to separate the logic of normal ingestion and ingest behind, i.e., put into different functions. But it is up to you.